### PR TITLE
Wire OverridePragueTime into txpool

### PR DESCRIPTION
**Pull request by yperbasis**
also extract common logic from `(p *TxPool) isShanghai()`, `isCancun()`, `isPrague()`.

### CODE REVIEW

**Comment by yperbasis** [comment]
Let's merge PR #11235 first

**Comment by sudeepdino008** [comment]
is the `OverridePragueTime` config something we eventually remove (asking because I don't see corresponding overrides for existing forks).

File: erigon-lib/txpool/pool.go [hunk 1]
```diff
@@ -983,31 +983,34 @@ func requiredBalance(txn *types.TxSlot) *uint256.Int {
...
+		isPostFlag.Swap(true)
 	}
 	return activated
 }
 
+func (p *TxPool) isShanghai() bool {
+	return isTimeBasedForkActivated(&p.isPostShanghai, p.shanghaiTime)
+}
+
 func (p *TxPool) isAgra() bool {
```
Comment (sudeepdino008) [inline erigon-lib/txpool/pool.go:1014]: refactor isAgra to use isTimeBasedForkActivated?

Comment (sudeepdino008) [reply erigon-lib/txpool/pool.go:1014]: never mind. I see that it uses different logic

**Review by sudeepdino008 — APPROVED** [review]

**Pull request merged**